ui/components/component-library/button-primary/button-primary.scss

```diff
@@ -13,7 +15,9 @@
   &--type-danger:not(&--disabled) {
     &:hover {
       color: var(--color-error-inverse);
-      box-shadow: var(--component-button-danger-shadow);
+      box-shadow:
+        var(--shadow-size-sm)
+        var(--color-error-shadow);
     }
 
     &:active {
```

Comment:
Shadows for the `ButtonPrimary` component work as expected

### Before


https://github.com/MetaMask/metamask-extension/assets/8112138/a58db93b-48d5-4246-b4b2-d4edb266eaea

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/ca8ce6cb-b5fe-456a-ab86-fbcbe12d02ec